clean-code-challanges/src/main/java/Anagram.java

```diff
@@ -7,11 +10,27 @@
  */
 public class Anagram {
 
-  public Anagram(String word) {
+  private final String word;
 
+  public Anagram(String word) {
+    this.word = word;
   }
 
   public List<String> match(List<String> candidates) {
-    return null;
+    List<String> foundAnagrams = new ArrayList<>();
+    char[] wordChars = this.word.toLowerCase().toCharArray();
+
+    Arrays.sort(wordChars);
+
+    for (String candidate : candidates) {
+      char[] candidateChars = candidate.toLowerCase().toCharArray();
+      Arrays.sort(candidateChars);
```

Comment:
Da du sowieso zweimal sortieren musst, hätte ich dies in eine Methode umgewandelt z.b sortLetters(). So kannst du wiederholenden Code vermeiden. Zitat Marco Rensch: "Wiederholende Code isch schlecht!" ;-)